src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1843,11 +1843,9 @@ public final boolean isAlive() {
 
     /**
      * Returns true if this thread is alive.
-     * This method is non-final so it can be overridden by VirtualThread.
-     * This method needs to be synchronized to ensure that thread termination
-     * happens-before isAlive() returning false.
+     * This method is non-final so it can be overridden.
```

Comment:
```suggestion
     * This method is non-final so it can be overridden by VirtualThread.
```

Reply:
:) I actually made that change in an earlier commit but decided to revert it because I assumed (@AlanBateman can confirm or refute) that it was left unspecific in case classes other than VirtualThread needed to override it.

Reply:
isAlive is final so it delegates to an overridable method to allow for other implementations. Right now, it's just VirtualThread but the alt implementation might have to override it too. I think the comment can be left as it is.